snippets/product-variant-options.liquid

```diff
@@ -43,12 +45,47 @@
           endif
       endcase
     endfor
+
+    if value.swatch.image
+      assign image_url = value.swatch.image | image_url: width: 50
+      assign swatch_value = 'url(' | append: image_url | append: ')'
+    elsif value.swatch.color
+      assign swatch_value = 'rgb(' | append: value.swatch.color.rgb | append: ')'
+    else
+      assign swatch_value = nil
+    endif
   -%}
 
-  {%- if block.settings.picker_type == 'button' -%}
+  {%- capture input_id -%}
+    {{ section.id }}-{{ option.position }}-{{ forloop.index0 -}}
```

Comment:
This will break if you embed multiple form in a section (think of a listing of products with quick view, a shop the look section allowing to highlight multiple products)... This emphasize the importance of my proposal here: https://github.com/Shopify/dawn/issues/2483